src/core/src/runtime/tensor_util.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/runtime/tensor_util.hpp"
+
+#include <filesystem>
+#include <fstream>
+
+#include "openvino/runtime/aligned_buffer.hpp"
+#include "openvino/runtime/shared_buffer.hpp"
+#include "openvino/util/file_path.hpp"
+#include "openvino/util/mmap_object.hpp"
+
+namespace ov {
+
+namespace {
+
+class StaticBufferAllocator {
+    std::shared_ptr<ov::AlignedBuffer> buffer;
+    bool allocated = false;  // if buffer was returned as allocated region
+
+public:
+    StaticBufferAllocator(std::shared_ptr<ov::AlignedBuffer> _buffer) : buffer(_buffer) {}
+
+    void* allocate(const size_t bytes, const size_t alignment) {
+        OPENVINO_ASSERT(alignment == alignof(max_align_t) || alignment == 0);
+        OPENVINO_ASSERT(!allocated);
+        OPENVINO_ASSERT(bytes == buffer->size());
+        allocated = true;
+        return buffer->get_ptr();
+    }
+
+    void deallocate(void* handle, const size_t bytes, const size_t alignment) {}
+
+    bool is_equal(const StaticBufferAllocator&) const {
+        return true;
+    }
+};
+}  // namespace
+
+void save_tensor_data(const ov::Tensor& tensor, const std::filesystem::path& file_name) {
+    OPENVINO_ASSERT(tensor.get_element_type() != ov::element::string);
+    const char* data = reinterpret_cast<char*>(tensor.data());
+    std::ofstream file(file_name, std::ios::binary);
+    file.write(data, tensor.get_byte_size());
+}
+void save_tensor_data(const ov::Tensor& tensor, const std::string& file_name) {
+    save_tensor_data(tensor, std::filesystem::path(file_name));
+}
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
+void save_tensor_data(const Tensor& tensor, const std::wstring& file_name) {
+    save_tensor_data(tensor, std::filesystem::path(file_name));
```

Comment:
This conversion can not always gives correct results. there is file_util to make it correct.
But keep only version with 'path' type and literal string will be converted to path or some additional util can be used if there will an issues. 
The api of these helpers will be simpler.

Reply:
only std::filesystem::path kept